### PR TITLE
vmm: broke up vstate in 2 separate files

### LINKED ISSUE #1605: Aesthetics: split src/vmm/src/vstate.rs in multiple files
Suggested option: split between `system`, `vm` and `vcpu` related functionality.

### PR DESCRIPTION
**Pull request by dianpopa**
## Reason for This PR

Fixes #1605.

## Description of Changes

vstate.rs is now a module with 2 submodules: vm.rs and vcpu.rs.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: src/vmm/src/vstate/vm.rs [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+        }
+    }
+}
+
+pub type Result<T> = result::Result<T, Error>;
+
+/// Describes a KVM context that gets attached to the microVM.
+/// It gives access to the functionality of the KVM wrapper as
+/// long as every required KVM capability is present on the host.
+pub struct KvmContext {
```
Comment (andreeaflorescu) [inline src/vmm/src/vstate/vm.rs:111]: This needs to be defined in another file probably. It is not related to the vm, but to the system wide state. In rust-vmm/kvm-ioctls, this functionality is part of the `system` module.

Comment (dianpopa) [reply src/vmm/src/vstate/vm.rs]: @andreeaflorescu  I applied your suggestion. Please re-review.

**Comment by andreeaflorescu** [comment]
nit: commit message should be updated as well. it is still mentioning two files.

**Comment by dianpopa** [comment]
@andreeaflorescu @georgepisaltu tests have finally passed. Please give it another look, thanks!

**Pull request merged**